apps/api/src/app/inbox/usecases/update-preferences/update-preferences.usecase.ts

```diff
@@ -0,0 +1,151 @@
+    };
+  }
+
+  private async createUserPreference(command: UpdatePreferencesCommand, subscriber: SubscriberEntity): Promise<void> {
+    const channelObj = {
+      chat: command.chat,
+      email: command.email,
+      in_app: command.in_app,
+      push: command.push,
+      sms: command.sms,
+    } as Record<ChannelTypeEnum, boolean>;
+
+    this.analyticsService.mixpanelTrack(AnalyticsEventsEnum.CREATE_PREFERENCES, '', {
+      _organization: command.organizationId,
+      _subscriber: subscriber._id,
+      _workflowId: command.workflowId,
+      level: command.level,
+      channels: channelObj,
+      __source: 'UpdatePreferences',
+    });
+
+    await this.subscriberPreferenceRepository.create({
+      _environmentId: command.environmentId,
+      _organizationId: command.organizationId,
+      _subscriberId: subscriber._id,
+      enabled: true,
+      channels: channelObj,
+      level: command.level,
+      ...(command.level === PreferenceLevelEnum.TEMPLATE && command.workflowId && { _templateId: command.workflowId }),
+    });
+  }
+  private async updateUserPreference(command: UpdatePreferencesCommand, subscriber: SubscriberEntity): Promise<void> {
+    const channelObj = {
+      chat: command.chat,
+      email: command.email,
+      in_app: command.in_app,
+      push: command.push,
+      sms: command.sms,
+    } as Record<ChannelTypeEnum, boolean>;
+
+    // Remove undefined values from the object
+    const updatePayload = Object.entries(channelObj).reduce((acc, [key, value]) => {
+      if (value !== undefined) {
+        acc[key] = value;
+      }
+
+      return acc;
+    }, {} as Record<ChannelTypeEnum, boolean>);
+
+    this.analyticsService.mixpanelTrack(AnalyticsEventsEnum.UPDATE_PREFERENCES, '', {
+      _organization: command.organizationId,
+      _subscriber: subscriber._id,
+      _workflowId: command.workflowId,
+      level: command.level,
+      channels: updatePayload,
+    });
+
+    const updateFields = {};
+    for (const [key, value] of Object.entries(updatePayload)) {
+      if (value !== undefined) {
+        updateFields[`channels.${key}`] = value;
+      }
+    }
+
+    await this.subscriberPreferenceRepository.update(
+      {
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+        _subscriberId: subscriber._id,
+        level: command.level,
+        ...(command.level === PreferenceLevelEnum.TEMPLATE &&
+          command.workflowId && { _templateId: command.workflowId }),
+      },
+      {
+        $set: updateFields,
+      }
+    );
+  }
+
+  private async findPreference(command: UpdatePreferencesCommand, subscriber: SubscriberEntity) {
+    const query = {
+      _organizationId: command.organizationId,
+      _environmentId: command.environmentId,
+      _subscriberId: subscriber._id,
+      level: command.level,
+      ...(command.level === PreferenceLevelEnum.TEMPLATE && command.workflowId && { _templateId: command.workflowId }),
+    };
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
✏ **suggestion:** this query is a repeatable code, which could be reused in a few places: line 126, line 82